src/Interpreters/inplaceBlockConversions.cpp

```diff
@@ -257,18 +257,6 @@ static std::unordered_map<String, ColumnPtr> collectOffsetsColumns(
 
                     if (offsets_column->size() != current_offsets_column->size() && inside_variant_element)
                         offsets_column = offsets_column->size() < current_offsets_column->size() ? offsets_column : current_offsets_column;
-#ifndef NDEBUG
```

Comment:
I had to remove this check as now it's possible to have 2 Array columns with not equal offsets (and it's ok and won't lead to any error). The author of this check agreed to remove it